include/cppkafka/utils/backoff_performer.h

```diff
@@ -134,6 +148,8 @@ class CPPKAFKA_API BackoffPerformer {
             // Increase out backoff depending on the policy being used
             backoff = increase_backoff(backoff);
         }
+        // No more retries left or we have a terminal error.
+        throw ConsumerException(rt.error_);
```

Comment:
Ok will add variadic templates and change the last exception to `Exception` which only takes a string. However I don't see how we can circumvent not using ReturnType since we still need to detect when the functor ends because it was successful or because the error callback returned false. Perhaps an int like before with `0=success+abort, 1=fail+abort,-1=fail+continue`.

Reply:
Yeah, you're right. I'd rather add an enum for this though, using `int`s is not ideal as you have no clue what those mean out of context.

Reply:
Ok before you responded I made this other change which doesn't need any enums. Take a look please. If you still want me to use enums I will. But the last commit requires the least API changes.